Tools/machines/perlmutter-nersc/spack-perlmutter-cuda.yaml

```
@@ -0,0 +1,91 @@
+# This is a Spack environment file.
+#
+# This environment can be used to install all dependencies to build the manual
+# locally.
+#
+# Activating and installing this environment will provide all dependencies
+# that are needed for full-feature development.
+#   https://spack.readthedocs.io/en/latest/environments.html
+#
+# Inside the directory of this file
+#   spack env create -d /global/common/software/${proj%_g}/$USER/spack-envs/warpx-perlmutter-cuda spack-perlmutter-cuda.yaml
+#   spack env activate /global/common/software/${proj%_g}/$USER/spack-envs/warpx-perlmutter-cuda
+#   spack install  # only needed the first time
+#
+spack:
+  # the top part is using the official template to chain the E4S modules at NERSC
+  #   https://docs.nersc.gov/applications/e4s/spack_environments/
+  #   https://github.com/NERSC/spack-infrastructure/blob/main/spack-configs/perlmutter-user-spack/spack.yaml
+  config:
+    concretization: together
+    build_stage: $env/var/spack/stage
+    misc_cache: $env/var/spack/misc_cache
+    concretizer: clingo
+    install_tree: $env/opt/spack
+
+  # Perlmutter compiler and package preferences
+  include:
+  - /global/common/software/spackecp/perlmutter/spack_settings/compilers.yaml
+  - /global/common/software/spackecp/perlmutter/spack_settings/packages.yaml
+
+  mirrors:
+    perlmutter-spack-develop: file:///global/common/software/spackecp/mirrors/perlmutter-spack-develop
+    perlmutter-e4s-22.05: file:///global/common/software/spackecp/mirrors/perlmutter-e4s-22.05
+
+  # Spack Chaining, if you want to use existing software stack
+  upstreams:
+    perlmutter-e4s-22.05:
+      install_tree: /global/common/software/spackecp/perlmutter/e4s-22.05/default/spack/opt/spack
+
+  # these are the WarpX dependencies
+  specs:
+  # https://github.com/ornladios/ADIOS2/issues/3332
+  - adios2 ~cuda
+  - blaspp
+  - boost
+  - ccache
+  - cmake
+  - cuda
+  - hdf5
+  - lapackpp
```

@sleak-lbl found a way to fix the concretizer issue via:
```suggestion
  - lapackpp^blaspp+cuda
```

I might be able to move this down to the `packages` variant, too...